### PR TITLE
Use constants instead of hardcoding the ports

### PR DESCRIPTION
**Pull request by sridhargaddam**
Depends on https://github.com/submariner-io/submariner/pull/1859

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: cmd/subctl/cloud.go [hunk 1]
```diff
@@ -57,8 +60,13 @@ func init() {
...
-	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.Natt, "natt-port", 4500, "IPSec NAT traversal port")
-	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.NatDiscovery, "nat-discovery-port", 4490, "NAT discovery port")
+	defaultTunnelPort, _ := strconv.ParseInt(subv1.DefaultUDPPort, 0, 32)
+	defaultNatDiscoveryPort, _ := strconv.ParseInt(subv1.DefaultNATTDiscoveryPort, 0, 32)
+
+	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.Natt, "natt-port", uint16(defaultTunnelPort),
+		"IPSec NAT traversal port")
+	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.NatDiscovery, "nat-discovery-port",
+		uint16(defaultNatDiscoveryPort), "NAT discovery port")
 	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.Vxlan, "vxlan-port", 4800, "Internal VXLAN port")
```
Comment (sridhargaddam) [inline cmd/subctl/cloud.go]: I tried to use constants for vxlan-port but when i include the route-agent/kubeproxy constants file, its trying to import many other dependencies and is throwing errors. So, avoided making the change.

Comment (tpantelis) [reply cmd/subctl/cloud.go]: yeah [this PR](https://github.com/submariner-io/submariner/pull/1859) refactors these constants and others.

File: cmd/subctl/cloud.go [hunk 2]
```diff
@@ -57,8 +60,13 @@ func init() {
 	restConfigProducer.AddKubeContextFlag(cloudCmd)
 	rootCmd.AddCommand(cloudCmd)
 
-	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.Natt, "natt-port", 4500, "IPSec NAT traversal port")
-	cloudPrepareCmd.PersistentFlags().Uint16Var(&cloudPorts.NatDiscovery, "nat-discovery-port", 4490, "NAT discovery port")
+	defaultTunnelPort, _ := strconv.ParseInt(subv1.DefaultUDPPort, 0, 32)
+	defaultNatDiscoveryPort, _ := strconv.ParseInt(subv1.DefaultNATTDiscoveryPort, 0, 32)
```
Comment (tpantelis) [inline cmd/subctl/cloud.go]: This would be no longer needed with https://github.com/submariner-io/submariner/pull/1859.

**Review by Jaanki — APPROVED** [review]

**Comment by sridhargaddam** [comment]
@tpantelis pushed a PR to refactor the ports in submariner repo - https://github.com/submariner-io/submariner/pull/1859#pullrequestreview-998671801

I'll update this PR to use the new constants once Tom's PR gets merged.

**Comment by Jaanki** [comment]
@sridhargaddam I added a `Depends on` trigger in the main PR description that would automatically show the status of the dependent PR right here.

**Comment by sridhargaddam** [comment]
> @sridhargaddam I added a `Depends on` trigger in the main PR description that would automatically show the status of the dependent PR right here.

Thanks. But this PR requires some additional changes, hence I marked it as Draft instead of just adding Depends on.

**Comment by tpantelis** [comment]
`vendor/github.com/submariner-io/submariner-operator/pkg/discovery/network/calico.go:56:44: undefined: constants.NetworkPluginCalico`

The **submariner-operator** needs to bump the **submariner** version as well so we need to wait until https://github.com/submariner-io/submariner-operator/pull/2099 is merged.

**Comment by tpantelis** [comment]
> submariner-io/submariner-operator#2099

It's merged - just need to bump the **submariner-operator** version.

**Review by tpantelis — APPROVED** [review]

**Pull request merged**